pmap/rpc_prog.go

```diff
@@ -0,0 +1,159 @@
+package pmap
+
+import (
+	"github.com/prashanthpai/sunrpc"
+)
+
+const (
+	portmapProgNum     = 34123456 // GLUSTER_PMAP_PROGRAM
+	portmapProgVersion = 1        // GLUSTER_PMAP_VERSION
+)
+
+const (
+	gfPmapNull        = iota
+	gfPmapPortByBrick // GF_PMAP_PORTBYBRICK
+	gfPmapBrickByPort // GF_PMAP_BRICKBYPORT
+	gfPmapSignUp      // Don't use
+	gfPmapSignIn      // GF_PMAP_SIGNIN
+	gfPmapSignOut     // GF_PMAP_SIGNOUT
+)
+
+// GfPortmap is a type for GlusterFS Portmap RPC program
+type GfPortmap struct {
+	name        string
+	progNum     uint32
+	progVersion uint32
+	procedures  []sunrpc.Procedure
+}
+
+// NewGfPortmap returns a new instance of GfPortmap type
+func NewGfPortmap() *GfPortmap {
+	// rpc/rpc-lib/src/protocol-common.h
+	return &GfPortmap{
+		name:        "Gluster Portmap",
+		progNum:     portmapProgNum,
+		progVersion: portmapProgVersion,
+		procedures: []sunrpc.Procedure{
+			sunrpc.Procedure{
+				sunrpc.ProcedureID{ProgramNumber: portmapProgNum, ProgramVersion: portmapProgVersion,
+					ProcedureNumber: gfPmapPortByBrick}, "PortByBrick"},
+			sunrpc.Procedure{
+				sunrpc.ProcedureID{ProgramNumber: portmapProgNum, ProgramVersion: portmapProgVersion,
+					ProcedureNumber: gfPmapBrickByPort}, "BrickByPort"},
+			sunrpc.Procedure{
+				sunrpc.ProcedureID{ProgramNumber: portmapProgNum, ProgramVersion: portmapProgVersion,
+					ProcedureNumber: gfPmapSignIn}, "SignIn"},
+			sunrpc.Procedure{
+				sunrpc.ProcedureID{ProgramNumber: portmapProgNum, ProgramVersion: portmapProgVersion,
+					ProcedureNumber: gfPmapSignOut}, "SignOut"},
+		},
+	}
+}
+
+// Name returns the name of the RPC program
+func (p *GfPortmap) Name() string {
```

Comment:
These interfaces(Name, Number, Version, Procedures) are not adding any value, but every sunrpc program has to implement it. All the details already available in the return object of NewGfPortmap. Can sunrpc library modified to use program_object.Name instead of program_object.Name()?